### ISSUE TITLE
Brew installation doesn't work properly when there is a newer jdk installed

### ISSUE THREAD

**Issue by kaviththiranga** [ballerina-platform/ballerina-lang#20389]

**Comment by kaviththiranga** [comment]
AFAIR we have declared openjdk 8 as a dependency in brew formula. But somehow, Ballerina is not working in this case.

**Comment by rasika** [comment]
Issue is brew only check & resolves dependencies during the installation time. It doesn't control the environment / runtime. Need to add a custom check for the Formula to look for default jdk version. From that way, we can halt installation of ballerina when the default jdk is not jdk8.

**Comment by kaviththiranga** [comment]
shall we print a warning and proceed with installation? Then user is aware that he needs to switch default jdk to jdk8 post installation.

**Comment by rasika** [comment]
Yes. Let's add a warning and continue installation. `Ballerina is only supported in JDK 8 error` is helpful to switchback when user is in a different JDK version.
